### PR TITLE
Make MsoComponent code trim friendly

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This change converts the interfaces to match the same patterns we're using with other Com interfaces. Also yanks the trace statements for these.

I'm probably going to try and factor this stuff out into an opt-in feature.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/10881)

### CODE REVIEW

File: src/System.Private.Windows.Core/src/Windows/Win32/System/Com/WinFormsComWrappers.cs [hunk 1]
```diff
@@ -71,15 +71,28 @@ protected override void ReleaseObjects(IEnumerable objects)
         }
     }
 
+    internal static TReturnType UnwrapAndInvoke<TThis, TInterface, TReturnType>(TThis* @this, Func<TInterface, TReturnType> func)
```
Comment (lonitra) [inline src/System.Private.Windows.Core/src/Windows/Win32/System/Com/WinFormsComWrappers.cs:74]: nit: docs

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.ComponentManager.cs [hunk 2]
```diff
@@ -21,23 +20,23 @@ public sealed partial class Application
     ///   This class is not used when running inside the Visual Studio shell.
     ///  </para>
     /// </remarks>
-    private unsafe class ComponentManager : IMsoComponentManager
+    private unsafe class ComponentManager : IMsoComponentManager.Interface
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.ComponentManager.cs:23]: Do we need `IManagedWrapper` on this?

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Application.ComponentManager.cs:23]: No, we never expose it out. We just stub it in if we don't have one to attach to.

File: src/System.Private.Windows.Core/src/Windows/Win32/System/Com/WinFormsComWrappers.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Private.Windows.Core/src/Windows/Win32/System/Com/WinFormsComWrappers.cs:74]: I'll catch it in my next PR.

**Review by lonitra — APPROVED** [review]

**Pull request merged**